hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/messagevalidation/V2PayloadValidator.java

```diff
@@ -48,10 +48,11 @@ public void validate(Exchange exchange, String v2Message) throws ValidationFaile
 		validateSendingFacility(exchange, messageObj, accessToken, isPharmanetMode);
 		validateReceivingApp(exchange, messageObj);
 		validateReceivingFacility(exchange, messageObj);
-		// TODO <REVIEW> Do we need to populate fields in validator? Should this be done after validation is complete?
-		populateOptionalField(messageObj);
-		validatePhanrmanetMessageFormat(exchange, v2Message, messageObj, isPharmanetMode);
-		exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, 200);
+		// TODO  ADDRESSED Do we need to populate fields in validator? Should this be done after validation is complete?
+		// This call was required to populate fields for error response message. Updated method name to "populateFieldsForErrorResponse" and moved call to method that needs these fields  
+		validatePharmanetMessageFormat(exchange, v2Message, messageObj, isPharmanetMode);
+		// Moved this call to PopulateHeader class to keep it all together
+		//exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, 200);
```

Comment:
Commented out code should be removed. It's available in the history.

Reply:
Done